students/ChengLiu/session09/mailroom_final.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+
+"""
+Cheng Liu
+Mail-room Final
+"""
+
+# import sys
+import math
+
+
+def donors():
+    """Testing data"""
+    return [Donor("Test Name_1", [1, 2, 3]),
+            Donor("Test Name_2", [10, 20, 30]),
+            Donor("Test Name_3"),  # tuples to list
+            ]
+
+
+class Donor():
+    """class for donors"""
+
+    def __init__(self, name, donations=None):
+        self.name = name.strip().title()
+        if donations is None:
+            self.donations = []
+        else:
+            self.donations = list(donations)  # converting donations from tuple to list
+
+    @property
+    def donor(self):
+        return (self.name, self.donations)
+
+    @property
+    def donor_name(self):
+        return self.name
+
+    @property
+    def list_donations(self):
+        try:
+            return self.donations
+        except IndexError:
+            return None
+
+    @property
+    def last_donation(self):
+        try:
+            return self.donations[-1]
+        except IndexError:
+            return None
+
+    @property
+    def total_donations(self):
+        try:
+            return sum(self.donations)
+        except IndexError:
```

Comment:
sum() will not return an IndexError -- it will return zero for an empty list, though :-)